### PR TITLE
Problem: zpoller_remove() does not always fail when removing non-existent reader

### PR DESCRIPTION
**Pull request by paddor**
Solution: When using zlist-based implementation (not ZMQ_HAVE_POLLER),
check for existence first. Set errno = EINVAL and return -1 if it
doesn't exist.

### CODE REVIEW

File: src/zpoller.c [hunk 1]
```diff
@@ -174,8 +174,14 @@ zpoller_remove (zpoller_t *self, void *reader)
     else
         rc = zmq_poller_remove_fd (self->zmq_poller, *(SOCKET *) reader);
 #else
-    zlist_remove (self->reader_list, reader);
-    self->need_rebuild = true;
+    if (zlist_exists (self-reader_list, reader)) {
```
Comment (bluca) [inline src/zpoller.c]: @paddor typo here, missing ">"

**Comment by bluca** [comment]
Good catch, had the same problem with the lack of feedback from zlist_remove in zmsg.

File: src/zpoller.c (hunk 1 above)
Comment (paddor) [inline src/zpoller.c]: Oh geez. Thanks!

**Comment by paddor** [comment]
@bluca Just looked that up. Yours is a more performant solution. I'll file another PR.

**Pull request merged**